File: core/services/identity/domain/domain.go
// Package domain defines the core domain types for the identity service.
package domain

import (
	identityv1 "milton_prism/pkg/pb/gen/milton_prism/types/identity/v1"
	tokenv1 "milton_prism/pkg/pb/gen/milton_prism/types/token/v1"
)

// Aliases proto — single source of truth, no mapping layer.
type (
	User                = identityv1.User
	UsersFilter         = identityv1.UsersFilter
	UserState           = identityv1.UserState
	AuthorizationTokens = tokenv1.AuthorizationTokens
)

// Re-export commonly used enum values.
const (
	UserStateUnspecified = identityv1.UserState_USER_STATE_UNSPECIFIED
	UserStateActive      = identityv1.UserState_USER_STATE_ACTIVE
	UserStateSuspended   = identityv1.UserState_USER_STATE_SUSPENDED
	UserStateDeleted     = identityv1.UserState_USER_STATE_DELETED
)

File: python/services/identity/domain/domain.py
"""Domain type aliases — proto messages are the single source of truth (A.3, Canon §3).

Import from here, never directly from gen/. This makes the boundary explicit and
lets us update proto paths in one place if the generated layout changes.
"""
from __future__ import annotations

from milton_prism.types.identity.v1.user_pb2 import User, UsersFilter, UserState
from milton_prism.types.pagination.v1.pagination_pb2 import Pagination
from milton_prism.types.token.v1.token_pb2 import AuthorizationTokens, Token

__all__ = [
    "AuthorizationTokens",
    "Pagination",
    "Token",
    "User",
    "UserState",
    "UsersFilter",
]
